### PR TITLE
Docathon: schedule.py and remove deprecated buffer arg

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

 - Remove the buffer options in Schedule, Instruction and ScheduleComponent which has been deprecated since 0.11.0. 
 - Update Schedule docstrings with minor formatting improvements, rst formatting directives, minor wording improvements.
 - Update the pulse `__init__` file to show Instruction near Schedule

### CODE REVIEW

File: qiskit/pulse/schedule.py [hunk 1]
```diff
@@ -188,50 +192,46 @@ def _union(self, other: Tuple[int, ScheduleComponent]) -> 'Schedule':
...
                name: Optional[str] = None) -> 'Schedule':
-        r"""Return a new schedule with `schedule` inserted at the maximum time over
-        all channels shared between `self` and `schedule`.
+        r"""Return a new schedule with ``schedule`` inserted at the maximum time over
+        all channels shared between ``self`` and ``schedule``.
+
+        .. math::
 
-       $t = \textrm{max}({x.stop\_time |x \in self.channels \cap schedule.channels})$
+            t = \textrm{max}({x.stop\_time |x \in self.channels \cap schedule.channels})
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]: As discussed, the rendering of this is odd. Would be nice to figure this out but also maybe isn't a priority for this PR.

File: qiskit/pulse/schedule.py [hunk 2]
```diff
@@ -188,50 +192,46 @@ def _union(self, other: Tuple[int, ScheduleComponent]) -> 'Schedule':
...
 
         Args:
-            schedule: schedule to be appended
-            buffer: Whether to obey buffer when appending
-            name: Name of the new schedule. Defaults to name of self
+            schedule: Schedule to be appended.
+            name: Name of the new schedule. Defaults to name of self.
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]:
```suggestion
            name: Name of the new ``Schedule``. Defaults to name of ``self``.
```

File: qiskit/pulse/schedule.py [hunk 3]
```diff
@@ -293,11 +295,11 @@ def exclude(self, *filter_funcs: List[Callable],
     def _apply_filter(self, filter_func: Callable, new_sched_name: str) -> 'Schedule':
         """
         Return a Schedule containing only the instructions from this Schedule for which
-        filter_func returns True.
+        ``filter_func`` returns True.
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]:
```suggestion
        ``filter_func`` returns ``True``.
```

File: qiskit/pulse/schedule.py [hunk 4]
```diff
@@ -241,20 +241,21 @@ def filter(self, *filter_funcs: List[Callable],
                intervals: Optional[Iterable[Interval]] = None) -> 'Schedule':
         """
         Return a new Schedule with only the instructions from this Schedule which pass though the
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]:
```suggestion
        Return a new ``Schedule`` with only the instructions from 
        this ``Schedule`` which pass though the
```

File: qiskit/pulse/schedule.py [hunk 5]
```diff
@@ -293,11 +295,11 @@ def exclude(self, *filter_funcs: List[Callable],
...
         """
         Return a Schedule containing only the instructions from this Schedule for which
-        filter_func returns True.
+        ``filter_func`` returns True.
 
         Args:
-            filter_func: function of the form (int, ScheduleComponent) -> bool
-            new_sched_name: name of the returned Schedule
+            filter_func: Function of the form (int, ScheduleComponent) -> bool.
+            new_sched_name: Name of the returned Schedule.
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]:
```suggestion
            new_sched_name: Name of the returned ``Schedule``.
```

File: qiskit/pulse/schedule.py [hunk 6]
```diff
@@ -430,9 +432,12 @@ def __eq__(self, other: ScheduleComponent) -> bool:
         """Test if two ScheduleComponents are equal.
 
         Equality is checked by verifying there is an equal instruction at every time
-        in `other` for every instruction in this Schedule.
+        in ``other`` for every instruction in this Schedule.
```
Comment (DanPuzzuoli) [inline qiskit/pulse/schedule.py]:
```suggestion
        in ``other`` for every instruction in this ``Schedule``.
```

File: releasenotes/notes/channel-module-deprecations-in-pulse-ffc384b325b077f1.yaml [hunk 7]
```diff
@@ -1,9 +1,10 @@
 ---
 upgrade:
   - |
-    Channel ``buffer`` option was deprecated in Terra 0.11.0 and has now been
-    removed. To add a delay on a channel, specify it explicitly in your
-    Schedule with a Delay::
+    The pulse ``buffer`` option found in Channel and Schedule was deprecated
```
Comment (mtreinish) [inline releasenotes/notes/channel-module-deprecations-in-pulse-ffc384b325b077f1.yaml]: Only one minor thing would be to link to the class documentation when referencing them:

```suggestion
    The pulse ``buffer`` option found in :class:`qiskit.pulse.Channel` and :class:`qiskit.pulse.Schedule` was deprecated
```

**Pull request merged**